packages/memory-storage/src/resource-clients/request-queue.ts

```diff
@@ -152,6 +153,16 @@ export class RequestQueueClient extends BaseClient implements storage.RequestQue
         }
     }
 
+    private *requestKeyIterator(rqClient: RequestQueueClient): IterableIterator<string> {
+        for (let i = this.forefrontRequestIds.length - 1; i >= 0; i--) {
```

Comment:
I need a second thought here - this iteration feels a bit dodgy, as we're modifying the `this.forefrontRequestIds` (e.g. at [here](https://github.com/apify/crawlee/pull/2681/files#diff-d2cb55420a8d6e84c8c7e076df7eb96eb74243fdfb091f3836fe7646dc7337ecR212)) in the iteration step. 

However, since we're walking through the array from the back (and always possibly removing only the last item), the index math checks out? I'd rather express this invariant somewhere explicitly, but I don't really see a place where this would fit.

Reply:
What if we do the filtering after the collecting of the requests? That way this should never (?) cause issues